pkg/pod/entrypoint_lookup.go

```diff
@@ -0,0 +1,123 @@
+package pod
+
+import (
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/authn/k8schain"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	lru "github.com/hashicorp/golang-lru"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// ResolveEntrypoints looks up container image ENTRYPOINTs for all steps that
+// don't specify a Command.
+//
+// Images that are not specified by digest will be specified by digest after
+// lookup in the resulting list of containers.
+func ResolveEntrypoints(cache EntrypointCache, namespace, serviceAccountName string, orig []corev1.Container) ([]corev1.Container, error) {
+	steps := orig[:] // defensive copy
```

Comment:
I want to make it clear that these methods don't modify the slice that's passed in, just in case. I also don't want to have to manually copy all the fields I _don't_ care about to the output. So I first make a copy of the input, then modify and return it.

Another route would be to unabashedly modify the input slice (that would probably also be fine), and not return a new slice. We could do that too, but I'd prefer to do it in a separate PR if we want to.

Reply:
Talked offline, it wasn't necessary after all! 🎉